src/main/java/org/janelia/saalfeldlab/n5/GsonKeyValueN5Reader.java

```diff
@@ -61,6 +61,21 @@ default boolean datasetExists(final String pathName) throws N5Exception {
 		return getDatasetAttributes(pathName) != null;
 	}
 
+	@Override
+	default boolean blockExists(final String dataset, final long[] blockPosition) {
+		final DatasetAttributes datasetAttributes = getDatasetAttributes(dataset);
```

Comment:
Wondering if we should instead (or in addition) add
`blockExists(String name, DatasetAttributes datasetAttributes, long[] blockPosition`?

Reply:
Reason: `getDatasetAttributes` incurs overhead that, in many practical cases, can be avoided?

Reply:
I think that sounds like a good idea. We could potentially add `DatasetAttributes` to the cache too, but right now we don't (just cache the raw json tree)